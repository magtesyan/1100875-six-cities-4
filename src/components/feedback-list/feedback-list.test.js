import renderer from "react-test-renderer";
import React from "react";
import shortid from "shortid";

import FeedbackList from "./feedback-list.jsx";

const reviews = [
  {
    id: shortid.generate(),
    text: `Everything. It was really really nice to get away and feel so comfortable where we was. My kids and I had a great time.`,
    assessment: 5,
    name: `Francis`,
    date: new Date(2020, 0, 1)
  },
  {
    id: shortid.generate(),
    text: `We like the beach view.`,
    assessment: 4,
    name: `Marjorie`,
    date: new Date(2019, 1, 2)
  },
];

it(`Feedback List is rendered correctly`, () => {
  const tree = renderer
    .create(
        <FeedbackList
          reviews={reviews}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
