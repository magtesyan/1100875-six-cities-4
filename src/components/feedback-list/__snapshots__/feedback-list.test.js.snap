// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback List is rendered correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/1.png"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Alex
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Everything. It was really really nice to get away and feel so comfortable where we was. My kids and I had a great time.
        </p>
        <time
          className="reviews__time"
          dateTime="May 2019"
        >
          May 2019
        </time>
      </div>
    </li>
  </ul>
</section>
`;
